### PR TITLE
Refoctor backup controller based on kubebuilder

### PR DESCRIPTION
**Pull request by qiuming-best**
Signed-off-by: Ming <mqiu@vmware.com>

Thank you for contributing to Velero!

# Please add a summary of your change

# Does your change fix a particular issue?

Fixes #(issue)
#5026, Refoctor backup controller based on kubebuilder
# Please indicate you've done the following:

- [ ] [Accepted the DCO](https://velero.io/docs/v1.5/code-standards/#dco-sign-off). Commits without the DCO will delay acceptance.
- [ ] [Created a changelog file](https://velero.io/docs/v1.5/code-standards/#adding-a-changelog) or added `/kind changelog-not-required` as a comment on this pull request.
- [ ] Updated the corresponding documentation in `site/content/docs/main`.

### CODE REVIEW

**Review by ywk253100 — APPROVED** [review]

File: pkg/cmd/server/server.go [hunk 1]
```diff
@@ -619,61 +590,13 @@ func (s *server) runControllers(defaultVolumeSnapshotLocations map[string]string
...
-			controller: backupController,
-			numWorkers: defaultControllerWorkers,
-		}
-	}
-
 	// By far, PodVolumeBackup, PodVolumeRestore, BackupStorageLocation controllers
 	// are not included in --disable-controllers list.
 	// This is because of PVB and PVR are used by node agent DaemonSet,
 	// and BSL controller is mandatory for Velero to work.
-	enabledControllers := map[string]func() controllerRunInfo{
```
Comment (qiuming-best) [inline pkg/cmd/server/server.go:672]: `enabledControllers` now are been removed

File: pkg/cmd/server/server.go [hunk 2]
```diff
@@ -696,19 +619,10 @@ func (s *server) runControllers(defaultVolumeSnapshotLocations map[string]string
...
 
-	// Instantiate the enabled controllers. This needs to be done *before*
-	// the shared informer factory is started, because the controller
-	// constructors add event handlers to various informers, which should
-	// be done before the informers are running.
-	controllers := make([]controllerRunInfo, 0, len(enabledControllers))
-	for _, newController := range enabledControllers {
-		controllers = append(controllers, newController())
-	}
-
```
Comment (qiuming-best) [inline pkg/cmd/server/server.go:711]: not need anymore

File: pkg/cmd/server/server.go [hunk 3]
```diff
@@ -865,13 +801,6 @@ func (s *server) runControllers(defaultVolumeSnapshotLocations map[string]string
 	// by v2.0, the block from this line and including the `s.mgr.Start() will be
 	// deprecated, since the manager auto-starts all the caches. Until then, we need to start the
 	// cache for them manually.
-	for i := range controllers {
-		controllerRunInfo := controllers[i]
-		// Adding the controllers to the manager will register them as a (runtime-controller) runnable,
-		// so the manager will ensure the cache is started and ready before all controller are started
-		s.mgr.Add(managercontroller.Runnable(controllerRunInfo.controller, controllerRunInfo.numWorkers))
-	}
```
Comment (qiuming-best) [inline pkg/cmd/server/server.go:873]: not need anymore

File: pkg/cmd/server/server.go [hunk 4]
```diff
@@ -865,13 +801,6 @@ func (s *server) runControllers(defaultVolumeSnapshotLocations map[string]string
...
-
 	s.logger.Info("Server starting...")
 
 	if err := s.mgr.Start(s.ctx); err != nil {
```
Comment (qiuming-best) [inline pkg/cmd/server/server.go:806]: Do we still need `s.mgr.Start`? @ywk253100 @blackpiglet

File: pkg/controller/generic_controller.go [hunk 5]
```diff
@@ -1,146 +0,0 @@
-/*
```
Comment (qiuming-best) [inline pkg/controller/generic_controller.go:1]: `generic_controller.go` no need anymore

File: pkg/cmd/server/server.go (hunk 4 above)
Comment (blackpiglet) [reply pkg/cmd/server/server.go:806]: This one is needed.

**Comment by blackpiglet** [comment]
This is the basic E2E test cases result.

https://gist.github.com/blackpiglet/2a2176ec5913b1bdf1937cdda421ab75

**Pull request closed without merging**